File: sejin/highScoreKit/체육복.py
def solution(n, lost, reserve):
    answer = 0
    tmp =[1]*(n+2)
    tmp[0] = 0
    tmp[n+1] = 0
    for i in reserve:
        tmp[i] += 1
    for i in lost:
        tmp[i]-=1
        if tmp[i-1] == 2:
            tmp[i-1]-=1
            tmp[i]=+1
        elif tmp[i+1] == 2:
            tmp[i+1] -=1
            tmp[i]+=1
    for i in tmp:
        if i >0:
            answer +=1
            
    return answer

File: qkfcl9987/20200405/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
	int answer = n - lost.size();
	for (int i = 0; i < lost.size(); i++)
	{
		for (int j = 0; j < reserve.size(); j++)
		{
			if (lost[i] == reserve[j])
			{
				lost[i] = 0;
				reserve[j] = 0;
				answer++;
				break;
			}
		}
	}


	for (int i = 0; i < lost.size(); i++)
	{
		if (lost[i] == 0) continue;
		for (int j = 0; j < reserve.size(); j++)
		{
			if (reserve[j] == 0) continue;
			if (lost[i] == reserve[j] - 1 || lost[i] == reserve[j] + 1)
			{
				reserve[j] = 0;
				answer++;
				break;
			}
		}
	}

	return answer;
}